File: MVC_RequireJS/Models/DTO/TaskItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace MVC_RequireJS.Models.DTO
{
    public class TaskItem
    {
        public TaskItem()
        {
        }
        public TaskItem(int id, DateTime taskDate, string title, string desc, bool completed)
        {
            Id = id;
            TaskDate = taskDate;
            Title = title;
            Description = desc;
            Status = completed ? "Complete" : "Queued";
        }

        public int Id { get; set; }
        [Required]
        public DateTime TaskDate { get; set; }
        [Required]
        public string Title { get; set; }
        public string Description { get; set; }
        public string Status { get; set; }
    }
}

File: MVC_RequireJS/Scripts/app/objects/task-item.js
//TaskItem obj

define(["knockout", "jquery", "knockout.validation", "ko/extenders"], function (ko, $, validation, extenders) {

    var TaskItem = function (id, title, description, taskDate, status) {
        return ko.validatedObservable({
            Id: ko.observable(id || ""),
            Title: ko.observable(title || "").extend({ required: true }),
            Description: ko.observable(description || ""),
            TaskDate: ko.observable(taskDate).extend({ required: true }),
            Status: ko.observable(status || "")
        });
    };

  
    return TaskItem;
});